xonsh/history.py

```diff
@@ -110,7 +110,8 @@ def files(self, only_unlocked=False):
         """
         _ = self  # this could be a function but is intimate to this class
         # pylint: disable=no-member
-        xdd = os.path.abspath(builtins.__xonsh_env__.get('XONSH_DATA_DIR'))
+        xdd = built_ins.expand_path(
```

Comment:
@scopatz I could but not sure how will this solve the (unrelated?) travis build failure with python 3.4.

Seems like the least of evils, simplest and most portable fix is to avoid all imports and apply https://github.com/scopatz/xonsh/commit/16d2096e2b7958f9f8aab3882c385899e1389f07
It has no issues in travis and fixes the original problem.